sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -720,6 +740,30 @@ class Analyzer(
     }
   }
 
+  case class ResolveCatalogAndNamespace(catalogManager: CatalogManager)
+    extends Rule[LogicalPlan] with LookupCatalog {
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case UnresolvedNamespace(CatalogAndNamespace(catalog, ns)) =>
+        ResolvedNamespace(catalog.asNamespaceCatalog, ns)
+    }
+  }
+
+  case class ResolveCatalogAndTable(catalogManager: CatalogManager)
```

Comment:
instead of creating a new rule, can we just catch `UnresolvedTable` in `ResolveTables` and `ResolveRelations`?